controller/program/src/instructions/swap_exact_in.rs

```diff
@@ -0,0 +1,402 @@
+    let out_lst_decimals = RawMint::of_acc_data(accounts.get(*ix_prefix.out_lst_mint()).data())
+        .and_then(Mint::try_from_raw)
+        .map(|a| a.decimals())
+        .ok_or(INVALID_ACCOUNT_DATA)?;
+
+    let protocol_fee_transfer_accs = transfer_checked_ix_account_handle_perms(
+        NewTransferCheckedIxAccsBuilder::start()
+            .with_auth(*ix_prefix.pool_state())
+            .with_mint(*ix_prefix.out_lst_mint())
+            .with_src(out_pool_reserves)
+            .with_dst(*ix_prefix.protocol_fee_accumulator())
+            .build(),
+    );
+
+    let signers_seeds = &[
+        PdaSeed::new(POOL_STATE_ID.as_slice()),
+        PdaSeed::new(&[POOL_STATE_BUMP]),
+    ];
+
+    cpi.invoke_signed(
+        accounts,
+        &out_lst_token_program,
+        TransferCheckedIxData::new(quote.0.protocol_fee, out_lst_decimals).as_buf(),
+        protocol_fee_transfer_accs,
+        &[PdaSigner::new(signers_seeds)],
+    )?;
+
+    let out_lst_transfer_accs = transfer_checked_ix_account_handle_perms(
+        NewTransferCheckedIxAccsBuilder::start()
+            .with_auth(*ix_prefix.pool_state())
+            .with_mint(*ix_prefix.out_lst_mint())
+            .with_src(out_pool_reserves)
+            .with_dst(*ix_prefix.out_lst_acc())
+            .build(),
+    );
+
+    cpi.invoke_signed(
+        accounts,
+        &out_lst_token_program,
+        TransferCheckedIxData::new(quote.0.out, out_lst_decimals).as_buf(),
+        out_lst_transfer_accs,
+        &[PdaSigner::new(signers_seeds)],
+    )?;
+
+    // Sync SOL values for LSTs
+    lst_sync_sol_val(
+        accounts,
+        cpi,
+        pool_state,
+        lst_state_list,
+        args.inp_lst_index as usize,
+        inp_lst_mint,
+        inp_pool_reserves,
+        inp_calc_prog,
+        inp_svc_accs_suf_range,
+    )?;
+    lst_sync_sol_val(
+        accounts,
+        cpi,
+        pool_state,
+        lst_state_list,
+        args.out_lst_index as usize,
+        out_lst_mint,
+        out_pool_reserves,
+        out_calc_prog,
+        out_svc_accs_suf_range,
+    )?;
+
+    // TODO: Confirm that I do need to do `accounts.get(pool_state)` again, like in line 49?
+    // Orelse, I get borrow issues, and what if I don't get the updated values after sync_sol_val
+    let pool = unsafe { PoolState::of_acc_data(accounts.get(pool_state).data()) }
+        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidPoolStateData))?;
+
+    let final_total_sol_value = pool.total_sol_value;
+
+    if final_total_sol_value < start_total_sol_value {
+        return Err(Inf1CtlCustomProgErr(Inf1CtlErr::PoolWouldLoseSolValue).into());
+    }
+
+    Ok(())
+}
+
+#[allow(clippy::too_many_arguments)]
+fn lst_sync_sol_val<'acc>(
+    accounts: &mut Accounts<'acc>,
+    cpi: &mut Cpi,
+    pool_state: AccountHandle<'acc>,
+    lst_state_list: AccountHandle<'acc>,
+    lst_index: usize,
+    lst_mint: AccountHandle<'acc>,
+    lst_reserves: AccountHandle<'acc>,
```

Comment:
these 4 `AccountHandle`s here, they seem familiar... I wonder if there's a struct that groups them together :thinking:

Reply:
<img width="611" height="430" alt="image" src="https://github.com/user-attachments/assets/cbecc51d-6676-4844-8dcc-65e25da0e802" />

Reply:
also i thought abt it again, maybe not `program/src/svc.rs` since this is specific to sync-sol-value, not the sol-value-calculator programs. Probably next to `sync_sol_val_with_retval` in `instructions/sync_sol_value.rs` is fine